### ISSUE TITLE
Different guava versions in dependencies

### ISSUE THREAD

**Issue by Dimok74** [springfox/springfox#1833]
Version 2.7.0.

Output of maven-enforcer-plugin:
`  +-io.springfox:springfox-swagger2:2.7.0`
`    +-io.springfox:springfox-spi:2.7.0`
`      +-io.springfox:springfox-core:2.7.0`
`        +-com.google.guava:guava:18.0`

`  +-io.springfox:springfox-swagger2:2.7.0`
`    +-io.springfox:springfox-spring-web:2.7.0`
`      +-org.reflections:reflections:0.9.11`
`        +-com.google.guava:guava:20.0`

**Comment by dilipkrish** [comment]
Is this breaking existing behavior?

**Comment by Dimok74** [comment]
I use RELEASE as version of springfox and enforcer plugin. All my builds are failed. it is nessesary to add some stuff to all poms:
        <dependency>
            <groupId>io.springfox</groupId>
            <artifactId>springfox-swagger2</artifactId>
            <version>RELEASE</version>
            <exclusions>
                <exclusion>
                    <groupId>com.google.guava</groupId>
                    <artifactId>guava</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>com.google.guava</groupId>
            <artifactId>guava</artifactId>
            <version>RELEASE</version>
        </dependency>
But I don't need guava dependency in my project.

**Comment by dilipkrish** [comment]
Will try to fix it in the next release, however this is a problem of transitive dependencies and your solution for enforcing versions it may not be conducive to handling it easily.

**Comment by kramik1** [comment]
It breaks for me.  I am seeing the following on a spring framework project during startup when moving from 2.6.1 to 2.7.0.

Caused by: java.lang.NoSuchMethodError: com.google.common.math.IntMath.saturatedAdd(II)I

Intellij shows 2 versions of Guava as described above.  Rather than add the extra dependency I have just moved back to 2.6.1.  I am using tomcat 8.5.16.

**Comment by kramik1** [comment]
So update on my issue.  I was also having issues with springfox-ui and swagger-resources/ui path not being found.  After an hour of frustration I cleaned maven and rebuilt the entire project and it works correctly.  The blog I was reading to try this library out used 2.6.1 for the core and 2.4 for the UI.  I updated the UI to match the core and that might be where the NoSuchMethodError came from.  I just ran it again with 2.7.0 and I didn't have any issues. 

http://www.baeldung.com/swagger-2-documentation-for-spring-rest-api
